loopdb/loopin.go

```diff
@@ -181,5 +208,48 @@ func deserializeLoopInContract(value []byte) (*LoopInContract, error) {
 		return nil, err
 	}
 
+	contract.Label, err = deserializeLabel(r)
+	if err != nil {
+		return nil, err
+	}
+
 	return &contract, nil
 }
+
+// deserializeLabel attempts to read an optional label field, serialized with
+// its length followed by value. If no length field is present, the swap was
+// created before we started writing labels, so we return an empty label. If we
+// do have a length field present, we read the label out if it's length is non
+// zero.
+func deserializeLabel(r io.Reader) (string, error) {
+	// Try to read our label length. If we hit an EOF, there are no more
+	// fields present so we can just return an empty contract.
+	var labelLength uint8
+	err := binary.Read(r, byteOrder, &labelLength)
+	if err == io.EOF {
```

Comment:
This is clever, but this also means that we cannot extend the serialized contract anymore as there's no way to express an empty label right?

Reply:
An empty label would be written with a 0 length? 

Legacy: EOF
No label: [0]
With Label: [len][label]

If we wanted to extend, we would be able to just add another field with the same format?

Reply:
Yeah, the problem is more theoretical than practical, but if we were to add another lenght/value field then if the client didn't set the first one (the label), then we need to insert it on the fly (or migrate lazily) in order to be able to deserialize correctly.